src/tree/updater_gpu_hist.cu

```diff
@@ -1057,6 +1061,11 @@ class GPUHistMakerSpecialised{
       right_node_max_elements = (std::max)(
         right_node_max_elements, shard->ridx_segments[nidx_right].Size());
     }
+    
+    if (param_.distributed_dask) {
```

Comment:
This if statement is unnecessary, I think the all reduce call is safe to use on a single machine.